src/ElmVersion.hs

```diff
@@ -58,3 +61,22 @@ style_0_19_cannotExposeOpenListing elmVersion =
         Elm_0_18 -> False
         Elm_0_18_Upgrade -> False
         _ -> True
+
+fromFile :: FileStore f => FilePath -> Free f ElmVersion
+fromFile path =
+    do
+        let dir = takeDirectory (path)
+        elmPackageJson <- stat (path </> "elm-package.json")
+        case elmPackageJson of
+            IsFile ->
+                do
+                    elmJson <- stat (path </> "elm.json")
+                    return $ case elmJson of
+                        IsFile -> Elm_0_19
+                        _ -> Elm_0_18
+
+            _ | path == dir ->
+                return Elm_0_19
```

Comment:
I think this branch means that if there is not `elm-package.json`, and we've recursed out as far as we can, then give Elm_0_19.  Is that right?   I think it should probably return `Nothing` in this case and let the caller decide what to default to rather than hardcoding the default in this function -- is there anything I'm missing about this?

Reply:
Oh, was this done to only have to check one file per level as you move outward?

Reply:
Fixed in last version.